### ISSUE TITLE
`Installing dependencies for Lambda Function "XXX"` infinitely repeats

### ISSUE THREAD

**Issue by vporton** [JoinColony/colonyCDapp#244]
On
```
npm install
```

`Installing dependencies for Lambda Function "XXX"`
infinitely repeats with XXX going through a loop.

**Comment by rdig** [comment]
Please provide more details in order for us be able to actually debug this:

- What OS are you running
- What Node version ?
- What NPM version ?
- What's the actual lambda function that exhibits this ?

**Comment by vporton** [comment]
> * What OS are you running
Ubuntu 22.10
> * What Node version ?
> 
> * What NPM version ?
$ node --version
v16.10.0
$ npm --version
7.24.0

> * What's the actual lambda function that exhibits this ?

Several lamdas (probably 4-10 different lambdas in a loop).

**Comment by jakubcolony** [comment]
I can confirm the issue, it happens whenever there's a folder in `/amplify/backend/function` that is missing `src/package.json` (it tends to happen when you switch between branches and have some leftover untracked files).
If package.json is missing, `npm i` will look in the parent directories until it finds the root package.json, which specifies the `postinstall` script, causing the infinite loop.

Thanks for reporting @vporton

**Comment by vporton** [comment]
No, it happens without any "you switch between branches and have some leftover untracked files".
```sh
git clone git@github.com:JoinColony/colonyCDapp.git && cd colonyCDapp && nvm use && npm install
```

```
$ node --version; npm --version
v16.10.0
7.24.0
```

Commit `d958b8e42591da3be23b9d258d6e44747c25d791`.
